lib/Runtime/Types/DynamicObject.cpp

```diff
@@ -149,25 +149,21 @@ namespace Js
         return NewObject<DynamicObject>(recycler, type);
     }
 
-    bool DynamicObject::Is(Var aValue)
+    bool DynamicObject::IsBaseDynamicObject(Var aValue)
     {
-        return RecyclableObject::Is(aValue) && (RecyclableObject::UnsafeFromVar(aValue)->GetTypeId() == TypeIds_Object);
+        return VarIs<RecyclableObject>(aValue) && (UnsafeVarTo<RecyclableObject>(aValue)->GetTypeId() == TypeIds_Object);
     }
 
-    DynamicObject* DynamicObject::FromVar(Var aValue)
+    template <> bool VarIs<DynamicObject>(RecyclableObject* obj)
     {
-        RecyclableObject* obj = RecyclableObject::FromVar(aValue);
-        AssertMsg(obj->DbgIsDynamicObject(), "Ensure instance is actually a DynamicObject");
-        AssertOrFailFast(DynamicType::Is(obj->GetTypeId()));
```

Comment:
Where has this failfast ended up, if anywhere?

Reply:
In `VarTo` implementation, it does `AssertOrFailFast(VarIs<T>(obj))`

---
In reply to: [202403174](https://github.com/Microsoft/ChakraCore/pull/5456#discussion_r202403174) [](ancestors = 202403174)